### DISCUSSION TITLE
Difference between project-role "Guest" and "Task Runner"

### DISCUSSION THREAD

**Discussion by noraab** [semaphoreui/semaphore#1738] [Q&A]
Just tried Semaphore the first time and was stumbling about what the difference between the two roles "Guest" and "Task Runner" is.
Is there any doc about the differences?

**Comment by samerbahri98** [comment]
A Task Runner can run tasks (ansible playbooks)
A Guest cannot do that.
This is the distinguishing permission

**Reply by noraab** [reply]
This is what I thought as well. But during my tests today, a guest could also run tasks. Maybe this is a bug. I can provide screenshots tomorrow.

**Reply by samerbahri98** [reply]
That is interesting, certainly that is not the intended behavior (from the code at least).
https://github.com/ansible-semaphore/semaphore/blob/develop/db/ProjectUser.go
if you can send screenshots that would be helpful

**Reply by noraab** [reply]
Ok, this seems to be a UI-Bug. I can see the "Run"-Button, even If having the role "Guest", but when clicking on it, the API responds with a `403`-Error which is as expected.

![image](https://github.com/ansible-semaphore/semaphore/assets/39309050/b2437213-23d9-4a36-9453-bd8ddfaa941e)
![image](https://github.com/ansible-semaphore/semaphore/assets/39309050/172693aa-57e1-47a6-b214-7c734787eed0)

**Reply by samerbahri98** [reply]
I see. I think the best course of action is to create an issue for this.
